iis-core/src/test/java/eu/dnetlib/iis/core/SshBasedHdfsFileFetcher.java

```diff
@@ -0,0 +1,133 @@
+	
+	public SshBasedHdfsFileFetcher(String remoteHost, String remoteUser, String remoteUserDir) {
+		this.remoteHost = remoteHost;
+		this.remoteUser = remoteUser;
+		this.remoteUserDir = appendRemoteSeparatorIfMissing(remoteUserDir);
+	}
+	
+	
+	//------------------------ LOGIC --------------------------
+	
+	/**
+	 * Fetches file (or directory) from hdfs into target directory.
+	 * 
+	 * @return location of fetched file
+	 */
+	public File fetchFile(String hdfsPath, File targetDir) throws IOException {
+		
+		String filename = new File(hdfsPath).getName();
+		File localTargetFile = new File(targetDir, filename);
+		
+		SSHClient sshClient = new SSHClient();
+		
+		try {
+			sshClient.loadKnownHosts();
+			sshClient.connect(remoteHost);
+			sshClient.authPublickey(remoteUser);
+
+			checkIfFileExistsOnHdfs(sshClient, hdfsPath);
+
+			copyFromHdfsOnRemote(sshClient, hdfsPath, remoteUserDir);
+
+			downloadFromRemote(sshClient, remoteUserDir + filename, localTargetFile);
+
+			removeFromRemote(sshClient, remoteUserDir + filename);
+
+		} finally {
+			sshClient.close();
+		}
+		
+		return localTargetFile;
+	}
+	
+	
+	//------------------------ PRIVATE --------------------------
+	
+	private void checkIfFileExistsOnHdfs(SSHClient sshClient, String hdfsPath) throws IOException {
+		if (sshExec(sshClient, "hadoop fs -test -e " + hdfsPath, false) != 0) {
+			throw new FileNotFoundException("File " + hdfsPath + " not found on hdfs");
+		}
+	}
+	
+	private void copyFromHdfsOnRemote(SSHClient sshClient, String hdfsSource, String remoteTarget) throws IOException {
+		sshExec(sshClient, "hadoop fs -get " + hdfsSource + " " + remoteTarget);
+	}
+	
+	private void downloadFromRemote(SSHClient sshClient, String remoteSource, File localTarget) throws IOException {
+		
+		localTarget.getParentFile().mkdirs();
+		
+		sshClient.newSCPFileTransfer().download(remoteSource, localTarget.getAbsolutePath());
+	}
+	
+	private void removeFromRemote(SSHClient sshClient, String remotePath) throws IOException {
+		sshExec(sshClient, "rm -r " + remotePath);
+	}
+	
+	
+	private int sshExec(SSHClient sshClient, String command) throws IOException {
+		return sshExec(sshClient, command, true);
+	}
+	
+	private int sshExec(SSHClient sshClient, String command, boolean throwExceptionOnCommandError) throws IOException {
+		Session sshSession = null;
+		int exitStatus;
+		try {
+			sshSession = sshClient.startSession();
+
+			Command cmd = sshSession.exec(command);
+			
+			cmd.join(SSH_EXEC_TIMEOUT_IN_SEC, TimeUnit.SECONDS);
+			
+			exitStatus = cmd.getExitStatus();
+			if (exitStatus != 0 && throwExceptionOnCommandError) {
+				throw new RuntimeException("Error executing command: " + command 
+						+ "\n" + IOUtils.readFully(cmd.getErrorStream()).toString());
+			}
+		} finally {
+			if (sshSession != null) {
+				sshSession.close();
+			}
+		}
+		
+		return exitStatus;
+	}
+	
+	private String appendRemoteSeparatorIfMissing(String directoryPath) {
```

Comment:
RemoteSeparator suggests that it is a special kind of separator (some remote separator). I would just name this method: appendFilePathSeparatorIfMissing.